src/Fmt_ast.ml

```diff
@@ -2681,7 +2693,11 @@ and fmt_constructor_arguments c ctx pre args =
           (list lds "@,; " (fmt_label_declaration c ctx))
 
 and fmt_constructor_arguments_result c ctx args res =
-  let pre : _ format = if Option.is_none res then " of@ " else ":@ " in
+  let pre : _ format =
+    if Option.is_none res then " of@ "
+    else if c.conf.space_before_colon then " :@ "
+    else ":@ "
```

Comment:
I think this case is like for constraints: a mistake in the current version, so better to always have the space.

Reply:
done